icu4c/source/i18n/currpinf.cpp

```diff
@@ -218,10 +215,8 @@ void
 CurrencyPluralInfo::setPluralRules(const UnicodeString& ruleDescription,
                                    UErrorCode& status) {
     if (U_SUCCESS(status)) {
-        if (fPluralRules) {
-            delete fPluralRules;
-            fPluralRules = nullptr;
-        }
+        delete fPluralRules;
+        fPluralRules = nullptr;
```

Comment:
Optional: this assignment `fPluralRules = nullptr;` is redundant with the following line that just reassigns it.  Feel free to leave as-is if you think it increases readability.

Reply:
Thanks Shane. I will push another commit that address this, but I think I will leave the other LocalUResourceBundlePointers for now.